### DISCUSSION TITLE
도메인은 왜 뷰를 의존하면 안 될까요?

### DISCUSSION THREAD

**Discussion by nayonsoso** [2024-woowacourse-study/level-interview#12] [OOP]
.

**Comment by J-I-H-O** [comment]
먼저 두 영역은 서로 다른 책임을 가지고 있습니다. 도메인은 비즈니스 로직과 관련된 부분입니다. 반면 뷰는 UI를 다루는 부분으로, 각각의 역할을 분명히 하는 것이 중요합니다.

또한 도메인이 뷰에 의존하게 되면 테스트하기가 어려워집니다. 뷰와 관련된 요소들이 도메인의 단위 테스트에 영향을 줄 수 있기 때문입니다.

또한 도메인이 뷰와 강하게 결합하면 재사용성이 저하되고, 유연성이 감소합니다. 
뷰 로직이 변경될 때마다 도메인 로직도 수정해야 하게 되기 때문입니다.
또 도메인이 특정 뷰에 의존하게 된다면 다른 뷰에서는 같은 도메인 로직을 재사용하기 어려워집니다.

**Comment by 3Juhwan** [comment]
뷰는 상대적으로 변경이 자주 일어나는 계층이며, 뷰가 도메인에 의존한다면 뷰의 변경이 도메인에도 영향을 미칠 수 있습니다. 그러나 도메인이 뷰에 의존한다면, 도메인의 변경이 뷰에도 영향을 줄 수 있습니다. 따라서 뷰와 도메인은 서로 독립적으로 설계되어야 하며, 변경이 적은 도메인이 뷰에 의존하는 것이 더 유리합니다. 이렇게 설계하면 뷰의 변경이 도메인 계층에 영향을 미치지 않으므로 코드의 유지보수성이 향상됩니다.
